board.h

```diff
@@ -0,0 +1,87 @@
+#ifndef Board_Class_H
+#define Board_Class_H
+
+#include <iostream>
+#include <algorithm>
+#include <vector>
+#include "pieces.h"
+
+struct indices {
+    std::size_t i, j;
+};
+
+template <typename T>
+Piece* Create() { return new T(); }
+
+typedef Piece* (*CreateFn)();
+
+
+class Board {
+private:
+    // playfield is 10 wide 40 tall.
+    // cell called using y * 10 + x
+    std::vector<int> board;
+    std::vector<int> rowPop;
+    static const std::vector<CreateFn> pieceNames;
+    Piece* currPiece;
+    int refpoint;
+    int offset;
+    int currHeight;
+
+public:
+
+    Board();
+
+    int& operator[](int rowCol);
+
+    int& operator[](indices rowCol); // called with board({row, col})
+
+    void newPiece();
```

Comment:
dont mix camel case and snake case. Pick one and stick to that convention. Also make sure your function names start with verbs unless its constructor/destructor.